packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```
@@ -0,0 +1,857 @@
+            throw new Error(
+                `No Coordinator server endpoint found in Coordinator Registry for feeRecipientAddress: ${feeRecipientAddress}. Registry contract address: ${
+                    this.registryAddress
+                }`,
+            );
+        }
+        return coordinatorOperatorEndpoint;
+    }
+
+    private async _generateSignedZeroExTransactionAsync(
+        data: string,
+        senderAddress: string,
+    ): Promise<SignedZeroExTransaction> {
+        const normalizedSenderAddress = senderAddress.toLowerCase();
+
+        const transaction: ZeroExTransaction = {
+            salt: generatePseudoRandomSalt(),
+            signerAddress: normalizedSenderAddress,
+            data,
+            verifyingContractAddress: this.exchangeAddress,
+        };
+        const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+        const signature = await signatureUtils.ecSignHashAsync(
+            this._web3Wrapper.getProvider(),
+            transactionHash,
+            transaction.signerAddress,
+        );
+        return {
+            ...transaction,
+            signature,
+        };
+    }
+
+    private async _executeServerRequestAsync(
+        signedTransaction: SignedZeroExTransaction,
+        txOrigin: string,
+        endpoint: string,
+    ): Promise<CoordinatorServerResponse> {
+        const requestPayload = {
+            signedTransaction,
+            txOrigin,
+        };
+        const response = await fetchAsync(`${endpoint}/v1/request_transaction?networkId=${this.networkId}`, {
+            body: JSON.stringify(requestPayload),
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json; charset=utf-8',
+            },
+        });
+
+        const isError = response.status !== HTTP_OK;
+
+        let json;
+        try {
+            json = await response.json();
+        } catch (e) {
+            // ignore
```

Hm, this feel dangerous to me. When would this fail?

---

this would fail if the server doesn't return a json response. e.g. something unexpected happens on server and it returns 500. I've had this happen a few times in testing, usually an uncaught exception from typeorm. 

This method basically tries to wrap server responses in a predictable format. The `json` field is either useful details or defaults to undefined, but `isError` and `status` are always informative.

---

If it only happens when a non-200 is returned, I'd rather we do:

```
const json = isError ? undefined : await response.json();
```

That way, if we are expecting a body, and `.json()` throws, it won't be swallowed. Generally very cautious of swallowing thrown errors since a lot of bugs go undiscovered in this way.

---

we actually expect a body for 200 and 400 but that's easy to express. Changing it!